src/gei/Commands/GenerateArchiveCommand.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.Threading.Tasks;
+
+namespace OctoshiftCLI.GithubEnterpriseImporter.Commands
+{
+    public class GenerateArchiveCommand : Command
+    {
+        private readonly OctoLogger _log;
+        private readonly ISourceGithubApiFactory _sourceGithubApiFactory;
+        private readonly EnvironmentVariableProvider _environmentVariableProvider;
+
+        public GenerateArchiveCommand(OctoLogger log, ISourceGithubApiFactory sourceGithubApiFactory, EnvironmentVariableProvider environmentVariableProvider) : base("generate-archive")
+        {
+            _log = log;
+            _sourceGithubApiFactory = sourceGithubApiFactory;
+            _environmentVariableProvider = environmentVariableProvider;
+
+            Description = "Invokes the GitHub Migration API's to generate a migration archive";
+            Description += Environment.NewLine;
+            Description += "Note: Expects GH_PAT and GH_SOURCE_PAT env variables to be set. GH_SOURCE_PAT is optional, if not set GH_PAT will be used instead. This authenticates to source GHES API";
+
+            var githubURL = new Option<string>("--github-url")
+            {
+                IsRequired = false
+            };
+            var githubSourceOrg = new Option<string>("--github-source-org")
+            {
+                IsRequired = true
+            };
+            var githubSourceRepo = new Option<string>("--github-source-repo")
+            {
+                IsRequired = true
+            };
+            var verbose = new Option("--verbose")
+            {
+                IsRequired = false
+            };
+
+            AddOption(githubURL);
+            AddOption(githubSourceOrg);
+            AddOption(githubSourceRepo);
+            AddOption(verbose);
+
+            Handler = CommandHandler.Create<string, string, string, bool>(Invoke);
+        }
+
+        public async Task Invoke(string githubURL, string githubSourceOrg, string githubSourceRepo, bool verbose = false)
+        {
+            _log.Verbose = verbose;
+
+            _log.LogInformation("Generating Migration Archive...");
+            _log.LogInformation($"GITHUB SOURCE ORG: {githubSourceOrg}");
+
+            if (string.IsNullOrWhiteSpace(githubURL))
+            {
+                _log.LogInformation("GithubURL not provided, defaulting to https://api.github.com");
```

Comment:
```suggestion
                _log.LogInformation("GitHubURL not provided, defaulting to https://api.github.com");
```

Reply:
We did you `Github` everywhere so if we change this we need to change it everywhere. It looks like a tech-dept to me.

Reply:
yeah I realized that about a month into coding this, that I capitilized github wrong, I've just been sticking with "Github" across the code base to keep it consistent (possibly using "GitHub" for user displayed strings).